### DISCUSSION TITLE
How to get github user's login

### DISCUSSION THREAD

**Discussion by imadatyatalah** [nextauthjs/next-auth#1917] [Help] [answered]
How can I get user login's value in user.userName?

I'm using github provider.

Example:
```js
accessToken: "...",
expires: "...",
user: {
   email: "user email",
   image: "user image",
   name: "user's name",
   userName: "user's login"
}
```

**Comment by balazsorban44** [comment, marked as answer]
https://next-auth.js.org/configuration/callbacks

**Reply by imadatyatalah** [reply]
Can you explain more?

**Reply by balazsorban44** [reply]
Check out the `jwt` and `session` callbacks. This page should also be useful: https://next-auth.js.org/configuration/providers#oauth-providers

**Reply by tatimunizz** [reply]
Hi! I have the same question. These links didn't helped me very much. I spend the whole day reading the documentation but I could not find the solution to this problem.
Form my application it is crucial to have the user login (username). :/

**Comment by tatimunizz** [comment]
Hi! @imadatyatalah 
Maybe this will help you. It did help me.
https://github.com/nextauthjs/next-auth/discussions/670

**Reply by imadatyatalah** [reply]
thanks @tatimunizz
